Tasks/Common/PowershellHelpers/Helpers.ps1

```diff
@@ -24,7 +24,10 @@ function Invoke-ActionWithRetries
         $RetryableExceptions,
 
         [string]
-        $RetryMessage
+        $RetryMessage,
+
+        [scriptblock]
+        $TimeoutAction
```

Comment:
If you want to keep an action which is run when an exception happens, then it should be generic enough i.e it should handle exceptions other than timeout.
I still believe this is not the right approach.

Reply:
Cool, lets remove it.